src/Autofac/Core/Resolving/ResolveOperationBase.cs

```diff
@@ -212,7 +196,10 @@ public object GetOrCreateInstance(ISharingLifetimeScope currentOperationScope, R
 
             try
             {
-                _pipelineTracer?.RequestStart(this, requestContext);
+                if (DiagnosticSource.RequestDiagnosticsEnabled())
```

Comment:
Suggest saving this value as a boolean to avoid the additional test when the request succeeds/fails.

Reply:
Each diagnostic write now does `if(DiagnosticSource.IsEnabled() && DiagnosticSource.IsEnabled("eventname"))` before writing, which is basically a check to see if there are _any_ subscribers (a null check) prior to checking if a specific event is enabled. This should be roughly as fast as the null check we had before.